client/block.c

```diff
@@ -526,7 +527,17 @@ static int add_block_nolock(struct xdag_block *newBlock, xdag_time_t limit)
 				}
 				break;
 
-			case XDAG_FIELD_RESERVE0:
+			case XDAG_FIELD_REMARK:
+				if(tmpNodeBlock.remark) {
+					xdag_free(tmpNodeBlock.remark);
+				}
+				tmpNodeBlock.remark = NULL;
```

Comment:
You don't need to assign NULL if you assign result of malloc in the next line.

Reply:
Done.